application/client/src/app/ui/views/toolbar/chart/zoomer/state.ts

```diff
@@ -0,0 +1,230 @@
+import { ISearchMap } from '@platform/interfaces/interface.rust.api.general';
+import { Instance as Logger } from '@platform/env/logger';
+import { Session } from '@service/session';
+import { Chart, ChartDataset } from 'chart.js';
+import { StoredEntity } from '@service/session/dependencies/search/store';
+import { FilterRequest } from '@service/session/dependencies/search/filters/request';
+import { ViewContainerRef } from '@angular/core';
+import { IlcInterface } from '@service/ilc';
+import { ChangesDetector } from '@ui/env/extentions/changes';
+import { getChartService, Service } from '../service';
+
+export class State {
+    public isCursorVisible: boolean = true;
+
+    private _parent!: IlcInterface & ChangesDetector;
+    private _filters!: Chart;
+    private _session!: Session;
+    private _service!: Service;
+    private _vcRef!: ViewContainerRef;
```

Comment:
You can store `HTMLElement` and pass it into state from component... No need to manipulate in state with angular-component reference